File: examples/record_highlight_test.ts
/**
 * Record a highlight test screencast using TypeScript + Playwright.
 *
 * Usage:
 *     npx tsx examples/record_highlight_test.ts <output-dir> <html-path> <color> <animation-speed-ms>
 */

import { chromium } from "playwright";
import { Storyboard, HighlightStyle } from "../api/typescript-client/src/index";

async function record(outputDir: string, htmlPath: string, color: string, animationSpeedMs: number): Promise<void> {
  const videosDir = `${outputDir}/videos`;

  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext({
    viewport: { width: 1280, height: 720 },
    recordVideo: {
      dir: videosDir,
      size: { width: 1280, height: 720 },
    },
  });
  const page = await context.newPage();

  const style: HighlightStyle = { color, animationSpeedMs };
  const storyboard = new Storyboard(outputDir, page, {
    debugOverlay: true,
    highlightStyle: style,
  });
  await storyboard.init();

  await page.goto(`file://${htmlPath}`, { waitUntil: "load" });
  await page.waitForSelector("#target", { state: "visible" });

  await storyboard.beginNarration();
  const button = page.locator("#target");
  await storyboard.highlight(button);
  await storyboard.endNarration();

  await context.close();
  await browser.close();
}

const [outputDir, htmlPath, color, speedStr] = process.argv.slice(2);
if (!outputDir || !htmlPath || !color || !speedStr) {
  console.error("Usage: npx tsx examples/record_highlight_test.ts <output-dir> <html-path> <color> <animation-speed-ms>");
  process.exit(1);
}

record(outputDir, htmlPath, color, parseInt(speedStr, 10)).catch((err) => {
  console.error(err);
  process.exit(1);
});

File: examples/record_highlight_test.py
"""Record a highlight test screencast using Python + Playwright.

Usage:
    python examples/record_highlight_test.py <output-dir> <html-path> <color> <animation-speed-ms>
"""

from __future__ import annotations

import sys
from pathlib import Path

from playwright.sync_api import sync_playwright

from screencast_narrator.storyboard import HighlightStyle, Storyboard


def record(output_dir: Path, html_path: Path, color: str, animation_speed_ms: int) -> None:
    videos_dir = output_dir / "videos"
    videos_dir.mkdir(parents=True, exist_ok=True)
    style = HighlightStyle(color=color, animation_speed_ms=animation_speed_ms)

    with sync_playwright() as pw:
        browser = pw.chromium.launch(headless=True)
        context = browser.new_context(
            viewport={"width": 1280, "height": 720},
            record_video_dir=str(videos_dir),
            record_video_size={"width": 1280, "height": 720},
        )
        page = context.new_page()

        storyboard = Storyboard(output_dir, page, debug_overlay=True, highlight_style=style)

        page.goto(f"file://{html_path}", wait_until="load")
        page.wait_for_selector("#target", state="visible")

        storyboard.begin_narration()
        button = page.locator("#target")
        storyboard.highlight(button)
        storyboard.end_narration()

        context.close()
        browser.close()


if __name__ == "__main__":
    if len(sys.argv) < 5:
        print("Usage: python examples/record_highlight_test.py <output-dir> <html-path> <color> <animation-speed-ms>", file=sys.stderr)
        sys.exit(1)
    record(Path(sys.argv[1]), Path(sys.argv[2]), sys.argv[3], int(sys.argv[4]))
